hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ContainerImporter.java

```diff
@@ -76,13 +82,31 @@ public void importContainer(long containerID, Path tarFilePath,
       HddsVolume hddsVolume, CopyContainerCompression compression)
       throws IOException {
 
-    HddsVolume targetVolume = hddsVolume;
-    if (targetVolume == null) {
-      targetVolume = chooseNextVolume();
+    synchronized (this) {
```

Comment:
Both `importContainerProgress` and `containerSet` are now accessed/changed under lock here, but without lock elsewhere.  We should apply synchronization consistently.

Reply:
containerSet access need not be in synchronized block, so removed. But for importContainerProgress other places is also added to synchronized block.